src/App.tsx

```diff
@@ -1,223 +1,197 @@
+          setTodos(response);
+        })
+        .catch(() => {
+          setError(true);
+          setMessageError('Todos from server were not gotten');
+        });
+    }
   }, []);
 
+  useMemo(() => {
+    setVisibelTodos(() => (
+      todos.filter(todo => {
+        switch (sortFilter) {
+          case 'active':
+            return !todo.completed;
+
+          case 'completed':
+            return todo.completed;
+
+          default:
+            return true;
+        }
+      })));
+  }, [todos, sortFilter]);
+
+  useMemo(() => {
+    setCompletedTodos(() => todos.filter(todo => todo.completed));
+  }, [todos]);
+
+  const handleSubmit = async (event: React.FormEvent) => {
+    event.preventDefault();
+    if (!user) {
+      return;
+    }
+
+    if (newTodoTitle.trim().length === 0) {
+      setError(true);
+      setMessageError('Title can\'t be empty');
+
+      return;
+    }
+
+    setVisibelTodos(state => [...state, {
+      id: 0,
+      userId: user.id,
+      title: newTodoTitle,
+      completed: false,
+    }]);
+
+    try {
+      const isAdding = await createTodo({
+        userId: user.id,
+        title: newTodoTitle,
+        completed: false,
+      });
+
+      setTodos(state => [...state, isAdding]);
+    } catch (errorFromServer) {
+      setError(true);
+      setMessageError('Unable to add a todo');
+    }
+
+    setTitle('');
+  };
+
+  const handleRemoveTodo = async (removeTodoID: number) => {
+    setActiveTodoId(removeTodoID);
+
+    try {
+      await deleteTodo(removeTodoID);
+
+      setTodos(state => state.filter(todo => todo.id !== removeTodoID));
+    } catch (errorFromServer) {
+      setError(true);
+      setMessageError('Unable to delete a todo');
+    } finally {
+      setActiveTodoId(0);
+    }
+  };
+
+  const handlerRemoveComleted = () => {
+    completedTodos.forEach((todoComleted) => handleRemoveTodo(todoComleted.id));
+  };
+
   return (
     <div className="todoapp">
       <h1 className="todoapp__title">todos</h1>
 
       <div className="todoapp__content">
         <header className="todoapp__header">
-          <button
-            data-cy="ToggleAllButton"
-            type="button"
-            className="todoapp__toggle-all active"
-          />
-
-          <form>
+          {todos.length > 0
+            && (
+              <button
+                data-cy="ToggleAllButton"
+                type="button"
+                className={classNames(
+                  'todoapp__toggle-all',
+                  { 'is-active': visibelTodos.length === 0 },
+                )}
+              />
+            )}
+          <form onSubmit={handleSubmit}>
             <input
               data-cy="NewTodoField"
               type="text"
               ref={newTodoField}
               className="todoapp__new-todo"
               placeholder="What needs to be done?"
+              value={newTodoTitle}
+              onChange={() => setTitle(newTodoField.current?.value || '')}
             />
```

Comment:
disable the input until receiving the response from the API (use isAdding variable);
isAdding should have a boolean type

Reply:
i add new todo with loader until get answer from server. I think it is the right behavior

Reply:
You should also disable input. This is a task requirement